php-checks/src/main/java/org/sonar/php/checks/PerlStyleCommentsUsageCheck.java

```diff
@@ -23,19 +23,23 @@
 import org.sonar.plugins.php.api.tree.lexical.SyntaxTrivia;
 import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 
-@Rule(key = PerlStyleCommentsUsageCheck.KEY)
+@Rule(key = "S2046")
 public class PerlStyleCommentsUsageCheck extends PHPVisitorCheck {
 
-  public static final String KEY = "S2046";
   private static final String MESSAGE = "Use \"//\" instead of \"#\" to start this comment";
 
   @Override
   public void visitTrivia(SyntaxTrivia trivia) {
     super.visitTrivia(trivia);
 
-    if (trivia.text().charAt(0) == '#') {
+    String text = trivia.text();
+    if (text.charAt(0) == '#' && !isShebangLine(text)) {
       context().newIssue(this, trivia, MESSAGE);
     }
   }
 
+  private static boolean isShebangLine(String triviaText) {
+    return triviaText.startsWith("#!");
+  }
```

Comment:
You could also consider/check that shebang can only be present on the first line of a file, but that will probably not improve much the accuracy.

Reply:
Good point. I think I will leave it as it is.
IMO it would be the cleaner solution to have a separate check, that checks whether a shebang line is on the first line.
But tbh. I don't see that much value in this, as I don't believe many users would face this issue.